### DISCUSSION TITLE
root permission

### DISCUSSION THREAD

**Discussion by IoSonoAndreaZ** [pgale/lineage_15.1_Installation_TSV#8] [Q&A]
sorry for noob question, I don't deal with custom rom since decades. How to obtain root permissions for some app that requires it?

**Comment by pimpinpanther96** [comment]
install magisk. Use magisk to patch your boot.img file
copy that file over to your pc and follow the same steps used to flash recovery partition in QFIL, except instead of replacing the recovery image, replace boot

**Reply by crazyjim68** [reply]
The easiest way to do this is rename the APK file to .zip and adb sideload it in recovery

**Comment by datenzar** [comment]
Simplest solution for me from [HA Forum](https://community.home-assistant.io/t/my-take-on-a-home-dashboard-running-on-the-lenovo-thinksmart-view-and-managing-multiple-devices-running-fully-kiosk/718355/181?u=datenzar:
)
1. Download the Magisk apk to your computer: [Releases · topjohnwu/Magisk · GitHub](https://github.com/topjohnwu/Magisk/releases)
2. Then connect a USB to the TSV and in ADB and Fastboot++ enter `adb reboot recovery`
3. Once booted into recovery mode select Apply Update - ADB Update. Then in ADB on your computer, enter `adb sideload Magisk-v28.apk`
4. Ignore error about "Signature Verification Failed" and reboot device
